roomControl.cs

```diff
@@ -116,12 +116,16 @@ private void colorUpdate()
             }
         }
 
-        private void delete (int Dnumber )
+        private void delete (int idNumber )
         {
             using (var connection = new SqlConnection(ConnectionString))
             {
                 connection.Open();
-                SqlCommand sqlDelete = new SqlCommand("delete * from Room where ID =(" + Dnumber + ")", connection);
+               
+                string DeleteQuery = "delete roomColor, roomNo, howManyDay, checkIn, checkOut, ID from Room where ID = idNumber ";
+                SqlCommand sqlCommand = new SqlCommand(DeleteQuery, connection);
```

Comment:
customer kısmından müştesi kayıtlarını da silmeli miyim sizce?
ve bu kısım için mesela " kayıtlar 24 saat de bir yedeklenir" gibi opsiyonlar var. buna da çıkış yapıldıktan 24 saat sonra(bu proje için 10 saniye olabilir, belki.) kayıtlar kaydedilir ve silinir gibi bir şey eklesem nasıl olur sizce?

Reply:
Eger dogru anladiysam burada bir oda siliyorsun. Oncelikle eger oda ile musteri arasinda bir foreign key iliskisi varsa burada hata alirsin. Yoksa orphan denilen kayitlar olur elinde. Yani bir yere referans vermissin fakat orada ona karsilik bir kayit yok. Bu durumda anlamsiz bir veri olur elinde. Oncelikle oda silinebilmeli mi ona karar vermek lazim. Bence oda silinebilir bir sey olmamali. Eger silinmesi gerekiyorsa `IsDeleted` ve `DeletedDate` diye kolonlar acip aktifligini kaldirabilirsin. Odalari gosterirken de IsDeleted = false diye bakarsin. Boylece elinde karsiliksiz kayit olmaz.

Reply:
anlamlandırmadığım şeyler ve aklıma takılan birkaç soru var konuşmamızda konuşmak üzere not alsam  olur mu ?

Reply:
evet şuan foreign key ile bağlı değil. Çünkü dediğiniz gibi hata aldım. Ve onunla ilgili bir çözüm bulamayıp çıkmaza girince yine hoş olmayan bir çözüm bulmuşum anladığım kadarıyla. Yeniden bunun üstünde çalışmaya ve araştırmaya başlayacağım.

Karar verme noktasında benim aklımda silinmezse bunlar karıştır mı düşüncesi oluşuyor ;  yani yeni bir oda kaydı olduğunda ve çıkış yapmış birinin kaydı silinmediği için odaların hepsi dolu gözükecek. Peki bu oluşacak karışıklığı çözmek için nasıl bir araştırma yapabilirim? @zencifil

Reply:
O zaman bir de sunu dusun: Gecen yil hangi tarihlerde hangi odalar doluydu? Bunun raporunu nasil hazirlayabilirsin?

Reply:
Evet, bunları hazırlayabilmem için bu verileri elimde tutmam gerekiyor. 

Teşekkür ederim. Bu konuyla ilgili araştırmaya başladım. @zencifil